### DISCUSSION TITLE
Best way to send an attachment via email

### DISCUSSION THREAD

**Discussion by LindezaGrey** [pocketbase/pocketbase#4686] [Q&A] [answered]
Hi there,

i was experimenting with sending emails. I need to send an attachment and came up with the following solution (located in pb_hooks)

```js

routerAdd("GET", "/debug", (c) => {
    const receiver = "mymail@example.com"
    const attachment = $filesystem.fileFromPath("/pb/pb_hooks/images/logo.svg").reader.open();
    const message = new MailerMessage({
        from: {
            address: $app.settings().meta.senderAddress,
            name:    $app.settings().meta.senderName,
        },
        to:      [{address: receiver}],
        subject: "my subject",
        html:    "<h2>YOUR_HTML_BODY...</h2>",
        attachments: {"logo.svg": attachment}

    });

    $app.newMailClient().send(message);
})
```

I was wondering if using $filesystem.fileFromPath etc. was the right way of doing it. Also i could not find a similar example, so if somebody wants to do the same and finds this, it might be helpful :)

**Comment by ganigeorgiev** [comment, marked as answer]
The above should be OK, just make sure to call `attachment.close()` after sending the mail.

```js
routerAdd("GET", "/debug", (c) => {
    const receiver = "mymail@example.com"
    const attachment = $filesystem.fileFromPath("/pb/pb_hooks/images/logo.svg").reader.open();
    try {
        const message = new MailerMessage({
            from: {
                address: $app.settings().meta.senderAddress,
                name:    $app.settings().meta.senderName,
            },
            to:      [{address: receiver}],
            subject: "my subject",
            html:    "<h2>YOUR_HTML_BODY...</h2>",
            attachments: {"logo.svg": attachment}
        });

        $app.newMailClient().send(message);
    } catch(err) {
        $app.logger().error("failed to send email", "error", err)
    } finally {
        attachment.close();
    }
})
```

Also note that currently the `attachments` field is only for regular/non-inline files, aka. if you are planning to use "logo.svg" as cid img source it will not work and probably will show as file for download in the mail client (see https://github.com/pocketbase/pocketbase/discussions/4523). Additionally even if we had support for it some web mail clients still may refuse to show an inline/embedded attachment as image and because of that I usually recommend using direct image links instead.

**Reply by LindezaGrey** [reply]
Thank you very much! Yes that is exactly what I wanted to achieve - in my case sending a PDF as an attachment.
I will implement the closing logic.
BTW your support of pocketbase is unreal, faster than any hotline i know. Either you are a whole team or an AI :rofl: , great job!

**Reply by taogaetz** [reply]
love bump for pocketbase. i have a similar use case, need to email a pdf on record creation. 

i believe i will find success combining the approach above with the info here:
https://github.com/pocketbase/pocketbase/discussions/4899
